### PR TITLE
Adding MongoDB State Store Implementation

### PR DESCRIPTION
**Pull request by shalabhms**
Adding MongoDB State Store Implementation

# Description

Adding state store implementation for Dapr for MongoDB

## Issue reference

#50 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

File: state/mongodb/mongodb.go [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+		meta.readconcern = val
+	}
+
+	var err error
+	var t time.Duration
+	if val, ok := metadata.Properties[operationTimeout]; ok && val != "" {
+		t, err = time.ParseDuration(val)
+	}
+
+	if err != nil {
```
Comment (yaron2) [inline state/mongodb/mongodb.go]: The correct Go way to write this is:

```golang
if err != nil {
  return nil, errors.New("incorrect operationTimeout field from metadata")
}

meta.operationTimeout = t
```

**Comment by shalabhms** [comment]
> @shalabhms you need to run go mod vendor and push the changes.

Thanks @yaron2 . I am doing this now and will push the changes with changes you suggested.

**Comment by yaron2** [comment]
Ok, now you just need to run `go mod vendor` and push the changes.

**Comment by shalabhms** [comment]
> Ok, now you just need to run go mod vendor and push the changes.

yes -onto it now.

**Comment by yaron2** [comment]
Lint is failing, fix the following comments:

```
state/mongodb/mongodb.go:45:2   golint     const `connectionUriFormat` should be `connectionURIFormat`
state/mongodb/mongodb.go:40:39  unconvert  unnecessary conversion
```

**Comment by yaron2** [comment]
LGTM

**Pull request merged**